server/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -70,6 +75,28 @@ public static TranslogReader open(
         return new TranslogReader(checkpoint, channel, path, header);
     }
 
+    /**
+     * Closes current reader and creates new one with new checkoint and same file channel
+     */
+    TranslogReader closeIntoTrimmedReader(long belowTerm, long aboveSeqNo, ChannelFactory channelFactory) throws IOException {
+        ensureOpen();
+        if (getPrimaryTerm() < belowTerm && aboveSeqNo < checkpoint.maxSeqNo) {
```

Comment:
See https://github.com/elastic/elasticsearch/pull/30176#discussion_r187008011 , which I think you missed. This method is called close and I think it should always close. term based checking can be done on the translog layer

Reply:
nope, the check is in Translog - agree that ensureOpen check in TranslogReader can go.